consensus/src/aggregator.rs

```diff
@@ -23,100 +24,102 @@ pub struct Aggregator(
-            }
+        let weight = committee
+            .votes_for(signer)
+            .ok_or(AggregatorError::NotCommitteeMember)?;
+
+        let (aggr_sign, cluster) =
+            self.0.entry((msg_step, vote.clone())).or_default();
+
+        // Each committee has 64 slots.
+        //
+        // If a Provisioner is extracted into multiple slots, then he/she only
+        // needs to send one vote which can be taken account as a vote for all
+        // his/her slots.
+        // Otherwise, if a Provisioner is only extracted to one slot per
+        // committee, then a single vote is taken into account (if more votes
+        // for the same slot are propagated, those are discarded).
+        if cluster.contains_key(signer) {
+            return Err(AggregatorError::DuplicatedVote);
+        }
 
-            // Aggregate Signatures
-            if let Err(e) = aggr_sign.add(signature) {
-                error!("{:?}", e);
-                return None;
-            }
+        // Aggregate Signatures
+        aggr_sign.add(signature)?;
+
+        // An committee member is allowed to vote only once per a single
+        // step. Its vote has a weight value depending on how many times it
+        // has been extracted in the sortition for this step.
+        let weight = cluster.set_weight(signer, weight);
+        debug_assert!(weight.is_some());
+
+        let total = cluster.total_occurrences();
+        let quorum_target = committee.super_majority_quorum();
+
+        debug!(
+            event = "vote aggregated",
+            %vote,
+            from = signer.to_bs58(),
+            added = weight,
+            total,
+            target = quorum_target,
+            signature = to_str(signature),
+        );
 
-            // An committee member is allowed to vote only once per a single
-            // step. Its vote has a weight value depending on how many times it
-            // has been extracted in the sortition for this step.
-            let weight = cluster.set_weight(signer, weight);
-            debug_assert!(weight.is_some());
+        let aggregate_signature = aggr_sign
+            .aggregated_bytes()
+            .expect("Signature to exist after aggregating");
+        let bitset = committee.bits(cluster);
 
-            let total = cluster.total_occurrences();
-            let quorum_target = committee.quorum();
+        let step_votes = StepVotes::new(aggregate_signature, bitset);
 
-            debug!(
-                event = "vote aggregated",
+        let quorum_reached = match &vote {
+            Vote::Valid(_) => total >= committee.super_majority_quorum(),
+            _ => total >= committee.majority_quorum(),
+        };
+
+        if quorum_reached {
+            tracing::info!(
+                event = "quorum reached",
                 %vote,
-                from = signer.to_bs58(),
-                added = weight,
                 total,
                 target = quorum_target,
```

Comment:
Isn't this `quorum_target` always equal to `committee.super_majority_quorum();` while `quorum_reached`  can be true due to exceeding `committee.majority_quorum(),`

Maybe we should have a specific  `quorum reached` log entry based on the type of the quorum.

Reply:
Agree with you, I'm ok for changing it to the current target